dependencies.yaml

```diff
@@ -395,6 +395,7 @@ dependencies:
           - distributed==2023.3.2.1
       - output_types: conda
         packages:
+          - cupy>=12.0.0
```

Comment:
This is coming in separately, right?

Reply:
Maybe not and this is it.

Reply:
I think rebasing this branch/merging latest 23.06 after the merge of #13284 should help clarify exactly what's new in this PR.